### PR TITLE
Start encoding after worker is ready

### PR DESCRIPTION
**Pull request by Kukunin**
It took about 100ms for the worker to initialize
For this time the main page sends a couple of encoding requests to worker
Which it starts to execute in order. This way sound latency is increased in streaming

### CODE REVIEW

**Comment by Kukunin** [comment]
Yes, it decreases. Realize:
- We take a buffer from the mic at time 0ms
- We tell worker to encode this buffer, but worker is not initialized
- We take a buffer from the mic at time 20ms
- We tell worker to encode this buffer, but worker is not initialized
....
- Worker initializes at time 100ms and encodes the first buffer, which was recorded 100ms ago
- Playback on the other end starts playing a buffer was recorded at least 100ms ago

After this PR, you start mic capturing only after worker is ready, so there is no this delay

**Comment by Kukunin** [comment]
Actually, recording starts before the worker is initialized, but recorded buffers discard

**Comment by chris-rudmin** [comment]
@Kukunin If I understand correctly, with this change the time from `rec.start()` to receiving the first page of audio data will be the same as before. However, In this scenario we are throwing away audio data (that the user might expect we are recording) until the worker is initialized.

File: src/waveWorker.js [hunk 1]
```diff
@@ -121,10 +122,9 @@ WavePCM.prototype.requestData = function(){
     wav.set( this.recordedBuffers[i], i * bufferLength + headerLength );
   }
 
-  global['postMessage']( wav, [wav.buffer] );
-  global['postMessage'](null);
+  global['postMessage']( {message: 'page', page: wav}, [wav.buffer] );
+  global['postMessage']( {message: 'page', page: null} );
```
[the remark this answers is not in the document]
Comment (Kukunin) [reply src/waveWorker.js]: I like this idea as well

**Comment by Kukunin** [comment]
Yes, you're right. We just discard recorded data unless the worker is initialized

**Comment by Kukunin** [comment]
Here I made it a little bit prettier. Can you test it please as well? I do streaming in my case, and it never ends =) thanks

File: src/recorder.js [hunk 2]
```diff
@@ -124,13 +124,31 @@ Recorder.prototype.initSourceNode = function( sourceNode ){
...
+          break;
+        case 'page':
+          onPage(e['data']['page']);
+          break;
+        case 'done':
+          this.finish();
+          break;
+      }
+    });
+    this.encoder.postMessage( Object.assign({
```
Comment (chris-rudmin) [inline src/recorder.js]: This will never be reached after the return statement.  Should this block go inside the `new Promise`?

Comment (chris-rudmin) [reply src/recorder.js]: Never mind. I did not read the code correctly.

**Pull request merged**